sonic_platform_base/chassis_base.py

```diff
@@ -0,0 +1,207 @@
+#
+# chassis_base.py
+#
+# Base class for implementing a platform-specific class with which
+# to interact with a chassis device in SONiC.
+#
+
+import sys
+from . import device_base
+
+
+class ChassisBase(device_base.DeviceBase):
+    """
+    Base class for interfacing with a platform chassis
+    """
+
+    # Possible reboot causes
+    REBOOT_CAUSE_POWER_LOSS = "power_loss"
+    REBOOT_CAUSE_THERMAL_OVERLOAD = "thermal_overload"
```

Comment:
Would it be possible to differentiate between thermal overload of Host CPU vs ASIC ?

Reply:
Addressed in https://github.com/Azure/sonic-platform-common/pull/13/commits/5fbf599a801665066545d8261ac3e3dfba34d762. Now differentiating between three types of thermal overloads, REBOOT_CAUSE_THERMAL_OVERLOAD_CPU, REBOOT_CAUSE_THERMAL_OVERLOAD_ASIC and REBOOT_CAUSE_THERMAL_OVERLOAD_OTHER.